cura/OneAtATimeIterator.py

```diff
@@ -1,149 +1,112 @@
-            self._node_stack = []
-            return
-
         node_list = []
         for node in self._scene_node.getChildren():
             if not issubclass(type(node), SceneNode):
                 continue
 
-            convex_hull = node.callDecoration("getConvexHull")
-            if convex_hull:
-                xmin = min(x for x, _ in convex_hull._points)
-                xmax = max(x for x, _ in convex_hull._points)
-                ymin = min(y for _, y in convex_hull._points)
-                ymax = max(y for _, y in convex_hull._points)
+            if node.callDecoration("getConvexHull"):
+                node_list.append(node)
 
-                convex_hull_polygon = Polygon.from_bounds(xmin, ymin, xmax, ymax)
-                if transform_x < 0:
-                    convex_hull_polygon = flip_x(convex_hull_polygon)
-                if transform_y < 0:
-                    convex_hull_polygon = flip_y(convex_hull_polygon)
 
-                node_list.append({"node": node,
-                                  "min_coord": [convex_hull_polygon.bounds[0], convex_hull_polygon.bounds[1]],
-                                  })
+        if len(node_list) < 2:
+            self._node_stack = node_list[:]
+            return
 
-        node_list = sorted(node_list, key = lambda d: d["min_coord"])
+        # Copy the list
+        self._original_node_list = node_list[:]
+
+        ## Initialise the hit map (pre-compute all hits between all objects)
+        self._hit_map = [[self._checkHit(i,j) for i in node_list] for j in node_list]
+
+        # Check if we have to files that block eachother. If this is the case, there is no solution!
+        for a in range(0,len(node_list)):
+            for b in range(0,len(node_list)):
+                if a != b and self._hit_map[a][b] and self._hit_map[b][a]:
+                    return
+
+        # Sort the original list so that items that block the most other objects are at the beginning.
+        # This does not decrease the worst case running time, but should improve it in most cases.
+        sorted(node_list, key = cmp_to_key(self._calculateScore))
+
+        todo_node_list = [_ObjectOrder([], node_list)]
+        while len(todo_node_list) > 0:
+            current = todo_node_list.pop()
+            for node in current.todo:
+                # Check if the object can be placed with what we have and still allows for a solution in the future
+                if not self._checkHitMultiple(node, current.order) and not self._checkBlockMultiple(node, current.todo):
+                    # We found a possible result. Create new todo & order list.
+                    new_todo_list = current.todo[:]
+                    new_todo_list.remove(node)
+                    new_order = current.order[:] + [node]
+                    if len(new_todo_list) == 0:
+                        # We have no more nodes to check, so quit looking.
+                        todo_node_list = None
```

Comment:
This value is unused. It always returns after setting this to `None`.

Reply:
Removed here: https://github.com/Ultimaker/Cura/pull/6434/commits/7bf2fa3b43d1ffe30af1f6161d37ceb8ed09d905